honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,311 @@
+    """Create a Ladybug Face3D object from a rhino3dm Brep.
+
+    This function is used in translating rhino3dm planar Brep to Ladybug Face3D objects.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    # * Check 01 - If any of the edge is an arc
+    lines = []
+    not_line = 0
+    for i in range(len(brep.Edges)):
+        if brep.Edges[i].IsLinear(tolerance) is False:
+            not_line += 1
+    # If there's an arc in one of the edges, mesh it
+    if not_line > 0:
+        warnings.warn(
+            "The surface has curved edges. It will be meshed.")
+        faces.extend(brep_to_face3d(brep))
+        return faces
+    else:
+        # Create Ladybug lines from start and end points of edges
+        for i in range(len(brep.Edges)):
+            start_pt = to_point3d(brep.Edges[i].PointAtStart)
+            end_pt = to_point3d(brep.Edges[i].PointAtEnd)
+            line = LineSegment3D.from_end_points(start_pt, end_pt)
+            lines.append(line)
+
+        # Create Ladybug Polylines from the lines
+        polylines = Polyline3D.join_segments(lines, tolerance)
+        # Sort the Ladybug Polylines based on length
+        # The longest polyline belongs to the boundary of the face
+        # The rest of the polylines belong to the holes in the face
+        sorted_polylines = sorted(polylines, key=lambda polyline: polyline.length)
+        # The first polyline in the list shall always be the polyline for
+        # the boundary
+        sorted_polylines.reverse()
+
+        # Getting all vertices of the face
+        mesh = brep.Faces[0].GetMesh(rhino3dm.MeshType.Any)
+        try:
+            mesh_pts = [to_point3d(mesh.Vertices[i])
+                        for i in range(len(mesh.Vertices))]
+        except AttributeError:
+            warnings.warn(
+                'Non-planar Breps are not handled by Honeybee. They are ignored.'
+            )
+            return faces
+        else:
+            # In the list of the Polylines, if there's only one polyline then
+            # the face has no holes
+            if len(sorted_polylines) == 1:
+                if len(mesh_pts) == 4 or len(mesh_pts) == 3:
+                    faces.extend(mesh_to_face3d(mesh))
+                else:
+                    boundary_pts = remove_dup_vertices(
+                        sorted_polylines[0].vertices, tolerance)
+                    lb_face = Face3D(boundary=boundary_pts)
+                    faces.append(lb_face)
+
+            # In the list of the Polylines, if there's more than one polyline then
+            # the face has hole / holes
+            elif len(sorted_polylines) > 1:
+                boundary_pts = remove_dup_vertices(
+                    sorted_polylines[0].vertices, tolerance)
+                hole_pts = [remove_dup_vertices(polyline.vertices, tolerance)
+                            for polyline in sorted_polylines[1:]]
```

Comment:
Again, you don't need to remove the duplicate vertices here.

Reply:
Below are two images. The first one is when duplicate vertices are removed. The second one is when they are not removed.
![duplicate vertices removed](https://user-images.githubusercontent.com/20086924/99557925-31f3f380-2991-11eb-8470-613ba9cf687c.jpg)
![duplicate vertices not removed](https://user-images.githubusercontent.com/20086924/99557953-38826b00-2991-11eb-9456-d973e60b3867.jpg)

Reply:
That's a Rhino display issue. I think you should just not remove vertices in your code and you can have a separate function that runs the `ladybug_geometry` `reomve_colinear_vertices` method if people want to have ALL of the duplicate vertices removed:

https://www.ladybug.tools/ladybug-geometry/docs/ladybug_geometry.geometry3d.face.html#ladybug_geometry.geometry3d.face.Face3D.remove_colinear_vertices

Reply:
I will try to once again demonstrate the issue that happens when I don't remove duplicate vertices.
This is a surface in Rhino with its vertices visible
![vertices](https://user-images.githubusercontent.com/20086924/99889004-43522f80-2c1f-11eb-9170-32c064ed2196.PNG)
In the image below, you can see the points for the boundary and the holes. Please also see the duplicate vertices at the end of each tuple of points.
![points](https://user-images.githubusercontent.com/20086924/99889012-595ff000-2c1f-11eb-86fa-5ddb83cdbbc1.PNG)
When I allow the duplicate vertices the outcome of the Honeybee face is the following;
![Honeybee Floor](https://user-images.githubusercontent.com/20086924/99889031-88766180-2c1f-11eb-85e8-306cc9e6e87f.PNG)

If this is a rhino issue then I need a way to fix it. Do I not? Removing duplicate vertices is the way I found. I am not aware of a better way but I can look for it if removing vertices is not acceptable.

Removal of duplicate vertices happens before I create a Face3D object. I am not sure how do you want me to use this `remove_colinear_vertices ` method for a Face3D object before creating a Face3D.

I have a feeling that I am missing something here.